snap/quota/resources.go

```diff
@@ -174,8 +176,8 @@ func (qr *Resources) validateJournalQuota() error {
 	}
 
 	if qr.Journal.Rate != nil {
-		if qr.Journal.Rate.Count <= 0 || qr.Journal.Rate.Period <= 0 {
-			return fmt.Errorf("journal quota must have a rate count and period larger than zero")
+		if qr.Journal.Rate.Count <= 0 || qr.Journal.Rate.Period.Microseconds() == 0 {
```

Comment:
If I compare with 0 it will compare against nanoseconds, and we do not support less than microseconds resolution (atleast systemd does not)

Reply:
yes, it merits a comment because it wasn't clear that was the intention of writing the check that way. Another approach would be to check  for  < time.Microsecond, that might be clearer without a comment?

Reply:
I think that makes a lot more sense, great suggestion!